src/transformation/visitors/optional-chaining.ts

```diff
@@ -121,26 +128,27 @@ export function transformOptionalChainWithCapture(
     // transform right expression first to check if thisValue capture is needed
     // capture and return thisValue if requested from outside
     let returnThisValue: lua.Expression | undefined;
-    const [rightPrecedingStatements, rightAssignment] = transformInPrecedingStatementScope(context, () => {
-        let result: lua.Expression;
-        if (thisValueCapture) {
-            ({ expression: result, thisValue: returnThisValue } = transformExpressionWithThisValueCapture(
-                context,
-                tsRightExpression,
-                thisValueCapture
-            ));
-        } else {
-            result = context.transformExpression(tsRightExpression);
+    const [rightPrecedingStatements, rightExpression] = transformInPrecedingStatementScope(context, () => {
+        if (!thisValueCapture) {
+            return context.transformExpression(tsRightExpression);
         }
-        return lua.createAssignmentStatement(luaTemp, result);
+
+        const { expression: result, thisValue } = transformExpressionWithThisValueCapture(
+            context,
+            tsRightExpression,
+            thisValueCapture
+        );
+        returnThisValue = thisValue;
+        return result;
     });
 
     // transform left expression, handle thisValue if needed by rightExpression
     const thisValueCaptureName = context.createTempName("this");
     const leftThisValueTemp = lua.createIdentifier(thisValueCaptureName, undefined, tempSymbolId);
     let capturedThisValue: lua.Expression | undefined;
 
-    const rightContextualCall = getOptionalContinuationData(tsTemp)?.contextualCall;
+    const optionalContinuationData = getOptionalContinuationData(tsTemp);
+    const rightContextualCall = optionalContinuationData?.contextualCall;
```

Comment:
I don't understand the point of this change

Reply:
optionalContinuationData is used later (line 193).